pkg/localstorev2/internal/chunkstamp/chunkstamp.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package chunkstamp
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"path"
+
+	"github.com/ethersphere/bee/pkg/localstorev2/internal"
+	"github.com/ethersphere/bee/pkg/postage"
+	"github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/hashicorp/go-multierror"
+)
+
+var (
+	// errMarshalInvalidChunkStampItemNamespace is returned during marshaling if the namespace is not set.
+	errMarshalInvalidChunkStampItemNamespace = errors.New("marshal chunkstamp.Item: invalid namespace")
+	// errMarshalInvalidChunkStampAddress is returned during marshaling if the address is zero.
+	errMarshalInvalidChunkStampItemAddress = errors.New("marshal chunkstamp.item: invalid address")
+	// errUnmarshalInvalidChunkStampAddress is returned during unmarshaling if the address is not set.
+	errUnmarshalInvalidChunkStampItemAddress = errors.New("unmarshal chunkstamp.item: invalid address")
+	// errMarshalInvalidChunkStamp is returned if the stamp is invalid during marshaling.
+	errMarshalInvalidChunkStampItemStamp = errors.New("marshal chunkstamp.item: invalid stamp")
+	// errUnmarshalInvalidChunkStampSize is returned during unmarshaling if the passed buffer is not the expected size.
+	errUnmarshalInvalidChunkStampItemSize = errors.New("unmarshal chunkstamp.item: invalid size")
+)
+
+var _ storage.Item = (*item)(nil)
+
+// item is the index used to represent a stamp for a chunk.
+//
+// Going ahead we will support multiple stamps on chunks. This item will allow
+// mapping multiple stamps to a single address. For this reason, the address is
+// part of the Namespace and can be used to iterate on all the stamps for this
+// address.
+type item struct {
+	// Keys.
+	namespace []byte // The namespace of other related item.
+	address   swarm.Address
+
+	// Values.
+	stamp swarm.Stamp
+}
+
+// ID implements the storage.Item interface.
+func (i *item) ID() string {
+	return path.Join(string(i.stamp.BatchID()), string(i.stamp.Index()))
+}
+
+// Namespace implements the storage.Item interface.
+func (i *item) Namespace() string {
+	return path.Join("chunkStamp", i.address.ByteString())
```

Comment:
I think you forgot to add the namespace here?

This should be prefixed by the namespace to avoid colliding between different namespaces.

Reply:
Done.